jumpscale/core/identity/__init__.py

```diff
@@ -156,10 +158,18 @@ def get_identity():
 
 class IdentityFactory(StoredFactory):
     _me = None
-
-    def new(
-        self, name, tname=None, email=None, words=None, explorer_url=None, tid=-1, admins=None,
-    ):
+    
+    def new(self, name, tname=None, email=None, words=None, explorer_url=None, tid=-1, admins=None,network=None):
+        if not words:
+            words = generate_mnemonic()
```

Comment:
we can check against the user in explorer using public key,

example here :
https://github.com/threefoldtech/js-ng/blob/419bef10c78e570f169b2ec18767d5749561d42f/jumpscale/threesdk/identitymanager.py#L138